lib/ruby_smb/server/server_client/session_setup.rb

```diff
@@ -80,12 +80,14 @@ def do_session_setup_smb2(request, session)
           if gss_result.nt_status == WindowsError::NTStatus::STATUS_SUCCESS
             session.state = :valid
             session.user_id = gss_result.identity
+            session.is_guest = !!gss_result.is_guest
             session.key = @gss_authenticator.session_key
-            session.signing_required = request.security_mode.signing_required == 1
+            session.signing_required = request.security_mode.signing_required == 1 || (!session.is_guest && !session.is_anonymous)
```

Comment:
Will this not force signing if its not explicitly enabled but the session is not a guest or anonymous session even if the server doesn't support signing?

Reply:
This is the server and the server supports signing. Signing is enabled if it's required or the user is not a guest or anonymous.